### PR TITLE
Update Developer Guide - Non Functional Requirements

### LINKED ISSUE #9: Update Developer Guide - Non Functional Requirements
[More details](https://nus-cs2103-ay2324s2.github.io/website/schedule/week7/project.html#4-update-the-dg-user-stories-glossary-nfrs-use-cases)

### PR DESCRIPTION
**Pull request by RyanNgWH**
Update developer guide to include non functional requirements for ClickConnect.

Closes #9.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -368,33 +370,35 @@ otherwise)
...
+1. Should work on any _mainstream OS_ (Windows, macOS and Linux) as long as it has Java `11` or above installed.
 2. Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be
+3. Should provide responsive performance, users should experience minimal delays in critical functionalities such as searching and updating contacts (feedback should be within 1 second).
+4. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be
    able to accomplish most of the tasks faster using commands than using the mouse.
-
-*{More to be added}*
+5. Should provide an intuitive and user-friendly interface. Users should be able to easily and quickly navigate the user interface to identify crucial information at a glance.
+6. Should have a shallow learning curve. New users (including the non technically savvy) should be able to pick up and start using the application efficiently within a week.
```
Comment (ruijietay) [inline docs/DeveloperGuide.md:395]:
```suggestion
6. Should have a shallow learning curve. New users (including the non-technically savvy) should be able to pick up and start using the application efficiently within a week.
```

Should `non technically` be hyphenated instead? *(i.e. `non-technically`)*

**Review by ruijietay — COMMENTED** [review]
Just a small typo, but apart from that, LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (RyanNgWH) [reply docs/DeveloperGuide.md]: You are right. Good catch!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -382,12 +382,16 @@ otherwise)
...
+1. Should work on any _mainstream OS_ (Windows, macOS and Linux) as long as it has Java `11` or above installed.
+1. Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+1. Should provide responsive performance, users should experience minimal delays in critical functionalities such as searching and updating contacts (feedback should be within 1 second).
+1. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be
+   able to accomplish most of the tasks faster using commands than using the mouse.
+1. Should provide an intuitive and user-friendly interface. Users should be able to easily and quickly navigate the user interface to identify crucial information at a glance.
+1. Should have a shallow learning curve. New users (including the non-technically savvy) should be able to pick up and start using the application efficiently within a week.
+1. Should be optimized to run smoothly on low-end devices with limited processing power and memory. Users on older hardware should be able to use the application as long as it meets [requirement 1](#nfr-1).
```
Comment (dillontkh) [inline docs/DeveloperGuide.md:394]: I think the list numbering might have been messed up in the process, but otherwise the content looks good!

**Review by dillontkh — CHANGES_REQUESTED** [review]
Minor fix required before approval.

**Review by dillontkh — APPROVED** [review]
LGTM!

**Pull request merged**